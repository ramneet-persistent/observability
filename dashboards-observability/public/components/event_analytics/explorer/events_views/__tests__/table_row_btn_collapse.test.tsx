/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { DocViewTableRowBtnCollapse } from '../detail_table/table_row_btn_collapse';

describe('DocViewTableRowBtnCollapse component', () => {
  configure({ adapter: new Adapter() });

  it('Renders DocViewTableRowBtnCollapse component collapsed', async () => {
    const wrapper = mount(<DocViewTableRowBtnCollapse onClick={jest.fn()} isCollapsed={false} />);

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
