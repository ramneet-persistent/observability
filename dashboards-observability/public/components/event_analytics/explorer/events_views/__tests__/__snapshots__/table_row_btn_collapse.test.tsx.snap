// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocViewTableRowBtnCollapse component Renders DocViewTableRowBtnCollapse component collapsed 1`] = `
<DocViewTableRowBtnCollapse
  isCollapsed={false}
  onClick={[MockFunction]}
>
  <EuiToolTip
    content="Toggle field details"
    delay="regular"
    position="top"
  >
    <span
      className="euiToolTipAnchor"
      onKeyUp={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <EuiButtonIcon
        aria-expanded={true}
        aria-label="Toggle field details"
        data-test-subj="collapseBtn"
        iconSize="s"
        iconType="arrowDown"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        <button
          aria-expanded={true}
          aria-label="Toggle field details"
          className="euiButtonIcon euiButtonIcon--primary euiButtonIcon--empty euiButtonIcon--xSmall"
          data-test-subj="collapseBtn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          type="button"
        >
          <EuiIcon
            aria-hidden="true"
            className="euiButtonIcon__icon"
            color="inherit"
            size="s"
            type="arrowDown"
          >
            <EuiIconEmpty
              aria-hidden={true}
              className="euiIcon euiIcon--small euiIcon--inherit euiIcon-isLoading euiButtonIcon__icon"
              focusable="false"
              role="img"
              style={null}
            >
              <svg
                aria-hidden={true}
                className="euiIcon euiIcon--small euiIcon--inherit euiIcon-isLoading euiButtonIcon__icon"
                focusable="false"
                height={16}
                role="img"
                style={null}
                viewBox="0 0 16 16"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              />
            </EuiIconEmpty>
          </EuiIcon>
        </button>
      </EuiButtonIcon>
    </span>
  </EuiToolTip>
</DocViewTableRowBtnCollapse>
`;
